crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -542,11 +541,18 @@ impl DiskFileBlobStoreInner {
         if from_disk.is_empty() {
             return Ok(res)
         }
+        let from_disk = from_disk
+            .into_iter()
+            .map(|(tx, data)| {
+                let data = Arc::new(data);
+                res.push((tx, data.clone()));
+                (tx, data)
+            })
+            .collect::<Vec<_>>();
```

Comment:
I think this makes sense, we're okay with doing this extra vec upfront if it reduces time(lock)